src/qt/guiutil.cpp

```diff
@@ -1013,43 +1015,92 @@ const std::vector<QString> listThemes()
+                QFileInfo info(*file);
+                QDateTime lastModified = info.lastModified(), prevLastModified;
+                auto it = mapLastModified.emplace(std::make_pair(*file, lastModified));
+                prevLastModified = it.second ? QDateTime() : it.first->second;
+                it.first->second = lastModified;
+                fModified = prevLastModified != lastModified;
+            }
+            return fModified;
+        };
 
-        QSettings settings;
-        QDir themes(":css");
-        QString theme = settings.value("theme", "").toString();
+        auto loadFiles = [&](const std::vector<QString>& vecFiles) -> bool {
+            if (fDebugCustomStyleSheets && !hasModified(vecFiles)) {
+                return false;
+            }
 
-        // Make sure settings are pointing to an existent theme
-        if (!isStyleSheetDirectoryCustom() && (theme.isEmpty() || !themes.exists(theme))) {
-            theme = defaultTheme;
-            settings.setValue("theme", theme);
-        }
+            stylesheet = std::make_unique<QString>();
 
-        auto loadFile = [&](const QString& name) {
-            QFile qFile(stylesheetDirectory + "/" + name + (isStyleSheetDirectoryCustom() ? ".css" : ""));
-            if (qFile.open(QFile::ReadOnly)) {
+            for (const auto& file : vecFiles) {
+                QFile qFile(file);
+                if (!qFile.open(QFile::ReadOnly)) {
+                    throw std::runtime_error(strprintf("%s: Failed to open file: %s", __func__, file.toStdString()));
+                }
                 stylesheet->append(QLatin1String(qFile.readAll()));
             }
+            return true;
+        };
+
+        auto pathToFile = [&](const QString& file) -> QString {
+            return stylesheetDirectory + "/" + file + (isStyleSheetDirectoryCustom() ? ".css" : "");
         };
 
+        std::vector<QString> vecFiles;
         // If light/dark theme is used load general styles first
         if (dashThemeActive()) {
-            loadFile("general");
 #ifndef Q_OS_MAC
-            loadFile("scrollbars");
+            vecFiles.push_back(pathToFile("scrollbars"));
 #endif
+            vecFiles.push_back(pathToFile("general"));
         }
+        vecFiles.push_back(pathToFile(getActiveTheme()));
 
-        loadFile(theme);
+        fStyleSheetChanged = loadFiles(vecFiles);
     }
 
-    return *stylesheet;
+    bool fUpdateStyleSheet = fDebugCustomStyleSheets && fStyleSheetChanged;
+
+    if (fDebugWidget) {
+        setWidgets.insert(widget);
+        QWidgetList allWidgets = QApplication::allWidgets();
+        auto it = setWidgets.begin();
+        while (it != setWidgets.end()) {
+            if (!allWidgets.contains(*it)) {
+                it = setWidgets.erase(it);
+                continue;
+            }
+            if (fUpdateStyleSheet && *it != widget) {
+                (*it)->setStyleSheet(*stylesheet);
+            }
+            ++it;
+        }
```

Comment:
why is this not a for loop?

Reply:
Don't you like `while` loops? :P Because it erases elements from the set and the `for` loop would still increment the iterator here and that doesn't make sense then. https://github.com/dashpay/dash/blob/e7ec855ff70dc5c9627337d6e9488d33669c2fb3/src/qt/guiutil.cpp#L1087

Reply:
Ah true. For a moment I thought a `continue` inside a for loop would result in skipping the increment, but obv. that isn't the case.